### DISCUSSION TITLE
I can't find how Widget UI demo got the permission for the web

### DISCUSSION THREAD

**Discussion by Zangana** [Canardoux/flutter_sound#602] [Q&A] [answered]
I can't find how the Widget UI demo got permission for the web. I looked deeply in the code still couldn't find how can you ask permission for the Web to record. On your live example, I can get permission to record but when I clone the Widget UI demo I can't get the permission. Your help will be appreciate

thank you.

**Comment by Larpoux** [comment]
startRecorder()  does that. You do not have to do it in your app

**Reply by Zangana** [reply]
Now I have these errors: 

`Error: Assertion failed: file:///D:/flutter_windows_v1.12.13+hotfix.8-stable/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_sound_web-7.5.3+1/lib/flutter_sound_recorder_web.dart:120:32
slotno == _slots.length`

` dart.global.newPlayerInstance is not a function
    at flutter_sound_player_web.FlutterSoundPlayerWeb.new.initializeMediaPlayer (http://localhost:52130/packages/flutter_sound_web/flutter_sound_player_web.dart.lib.js:76:42)`

**Comment by Larpoux** [comment, marked as answer]
You probably did not include the javascript files in the header of your HTML file.
Look [here](https://tau.canardoux.xyz/flutter_sound_install.html#flutter-web).

I think that this information should be duplicate in the Guides pages. I am going to do that.

**Reply by Zangana** [reply]
Thank you very much. It solved my problem.
